modules/SerialCommHub/tiny_modbus_rtu.cpp

```diff
@@ -262,19 +271,28 @@ bool TinyModbusRTU::open_device(const std::string& device, int _baud, bool _igno
 }
 
 int TinyModbusRTU::read_reply(uint8_t* rxbuf, int rxbuf_len) {
-    struct timeval timeout;
-    timeout.tv_sec = 0;
-    timeout.tv_usec = MODBUS_RX_INITIAL_TIMEOUT_MS * 1000; // 500msec intial timeout until device responds
+    // Lambda to convert std::chrono to timeval.
+    auto to_timeval = [](const auto& time) {
```

Comment:
This should be a stand alone function in the anonymous namespace in this file. It does not capture anything from its environment.

Reply:
I disagree - We can only use auto as a function argument in c++20. The the extracted function therefore would either need to pin down the exact chrono type (milliseconds) making the code slightly less generic or be a template function making the code less readable